### PR TITLE
feat(plugin-sdk): Add SchemaUiAnnotator and x-ui annotations for plugin UI forms

### PR DESCRIPTION
**Pull request by danielgerlag**
## Summary

Adds a `SchemaUiAnnotator` builder in `plugin-sdk` that injects `x-ui:*` extension properties into plugin JSON Schema definitions. These annotations drive schema-driven dynamic forms in the Drasi Server UI.

### New: `SchemaUiAnnotator` (`plugin-sdk/src/schema_ui.rs`)
Fluent builder API for injecting UI hints into serialized OpenAPI/JSON Schema:
```rust
SchemaUiAnnotator::new(&schemas)
    .field("host", |f| f.group("Connection").order(1).placeholder("localhost"))
    .field("password", |f| f.widget("password").group("Authentication"))
    .apply()
```

### Annotated all 6 plugins
- **postgres source**: Connection, Authentication, Tables, Replication, SSL groups
- **http source**: Connection, Polling, Authentication groups with widget hints
- **mock source**: Basic Data group
- **log reaction**: Output Settings
- **http reaction**: Connection, Authentication, Timeout groups
- **sse reaction**: Connection, Authentication, Event Configuration groups

## Supported x-ui extensions
| Extension | Purpose |
|-----------|---------|
| `x-ui:group` | Group fields into collapsible sections |
| `x-ui:order` | Control field ordering within groups |
| `x-ui:widget` | Override widget type (password, textarea, etc.) |
| `x-ui:placeholder` | Placeholder text for input fields |
| `x-ui:help` | Help text displayed below the field |
| `x-ui:condition` | Conditionally show/hide fields |
| `x-ui:collapsed` | Default collapsed state for groups |

## Companion PR
Requires [drasi-project/drasi-server PR](https://github.com/drasi-project/drasi-server/pulls) for the UI-side form rendering.

### CODE REVIEW

File: components/plugin-sdk/src/schema_ui.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+impl SchemaUiAnnotator {
+    /// Create a new annotator from a serialized schemas map JSON string.
+    ///
+    /// `root_schema_name` is the key in the map that identifies the root config schema
+    /// (e.g., `"source.postgres.PostgresSourceConfig"`).
+    pub fn new(schemas_json: &str, root_schema_name: &str) -> Self {
+        let schemas: Value =
+            serde_json::from_str(schemas_json).expect("SchemaUiAnnotator: invalid JSON input");
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs]: Fixed. `new()` now accepts `serde_json::Value` directly and returns `Result<Self, SchemaUiError>`. The `expect()` is gone — callers get a proper `SchemaUiError::RootSchemaNotFound` error. All 6 descriptor call sites updated.

File: components/plugin-sdk/src/schema_ui.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        let builder = builder_fn(FieldUiBuilder::new());
+        self.field_annotations
+            .push((field_name.to_string(), builder));
+        self
+    }
+
+    /// Apply all annotations and return the modified JSON string.
+    pub fn annotate(mut self) -> String {
+        if let Some(root) = self.schemas.get_mut(&self.root_schema_name) {
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs:192]: Fixed. new() now returns Err(SchemaUiError::RootSchemaNotFound) if the root schema name is missing. Also, annotate() fires a debug_assert! for unknown field names, catching typos immediately in dev/test builds.

File: components/plugin-sdk/src/schema_ui.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            .insert("x-ui:collapsed".to_string(), Value::Bool(collapsed));
+        self
+    }
+}
+
+/// Annotates an OpenAPI schema map with `x-ui:*` UI hint extensions.
+pub struct SchemaUiAnnotator {
+    schemas: Value,
+    root_schema_name: String,
+    field_annotations: Vec<(String, FieldUiBuilder)>,
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs:152]: Fixed. The expect() is gone -- new() now accepts serde_json::Value directly (no JSON parsing) and returns Result. The panic vector no longer exists.

File: components/plugin-sdk/src/schema_ui.rs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    pub fn condition_not_empty(mut self, field: &str) -> Self {
+        let mut condition = Map::new();
+        condition.insert("field".to_string(), Value::String(field.to_string()));
+        condition.insert("notEmpty".to_string(), Value::Bool(true));
+        self.annotations
+            .insert("x-ui:condition".to_string(), Value::Object(condition));
+        self
+    }
+
+    /// Set whether the group section starts collapsed.
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs]: Fixed. Updated the doc comment to clarify: "Sets x-ui:collapsed on this field, signalling that the group containing this field starts collapsed by default." Also documents the last-write-wins behavior when multiple fields in the same group set conflicting values.

File: components/plugin-sdk/src/schema_ui.rs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        F: FnOnce(FieldUiBuilder) -> FieldUiBuilder,
+    {
+        let builder = builder_fn(FieldUiBuilder::new());
+        self.field_annotations
+            .push((field_name.to_string(), builder));
+        self
+    }
+
+    /// Apply all annotations and return the modified JSON string.
+    pub fn annotate(mut self) -> String {
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs:191]: Fixed. new() now validates the root schema name upfront and returns Err(SchemaUiError::RootSchemaNotFound) if not found. Misconfiguration is no longer silent.

File: components/plugin-sdk/src/schema_ui.rs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    schemas: Value,
+    root_schema_name: String,
+    field_annotations: Vec<(String, FieldUiBuilder)>,
+}
+
+impl SchemaUiAnnotator {
+    /// Create a new annotator from a serialized schemas map JSON string.
+    ///
+    /// `root_schema_name` is the key in the map that identifies the root config schema
+    /// (e.g., `"source.postgres.PostgresSourceConfig"`).
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs:162]: Fixed. new() now accepts serde_json::Value directly. Callers use serde_json::to_value() instead of serde_json::to_string(), eliminating the unnecessary serialize-to-string then deserialize-back-to-Value roundtrip.

File: components/plugin-sdk/src/schema_ui.rs [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            .push((field_name.to_string(), builder));
+        self
+    }
+
+    /// Apply all annotations and return the modified JSON string.
+    pub fn annotate(mut self) -> String {
+        if let Some(root) = self.schemas.get_mut(&self.root_schema_name) {
+            if let Some(properties) = root.get_mut("properties") {
+                for (field_name, builder) in &self.field_annotations {
+                    if let Some(prop) = properties.get_mut(field_name) {
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs:195]: Fixed. Added 11 unit tests covering: happy path annotations, unknown field handling (with debug_assert!), missing root schema error, multiple annotations on same field, condition_value, condition_not_empty, collapsed, the new help() method, unannotated fields unchanged, and error display formatting.

File: components/plugin-sdk/src/schema_ui.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs]: Fixed. Updated the doc comment to: "Sets x-ui:collapsed on this field, signalling that the group containing this field starts collapsed by default. This is a group-level concept expressed on the field builder for convenience. If multiple fields in the same group set conflicting values, the last one written wins."

File: components/plugin-sdk/src/schema_ui.rs [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+    where
+        F: FnOnce(FieldUiBuilder) -> FieldUiBuilder,
+    {
+        let builder = builder_fn(FieldUiBuilder::new());
+        self.field_annotations
+            .push((field_name.to_string(), builder));
+        self
+    }
+
+    /// Apply all annotations and return the modified JSON string.
```
[the remark this answers is not in the document]
Comment (danielgerlag) [reply components/plugin-sdk/src/schema_ui.rs:187]: Fixed. The silent-no-op for missing root schema is gone -- new() now returns an error. The annotate() doc comment now also documents the debug_assert behavior for missing field names: "Fields named in .field() calls that do not exist in the root schema properties are silently skipped (a debug_assert! fires in debug builds)."

**Comment by danielgerlag** [comment]
All drasi-reviewer feedback has been addressed:

**🔴 Blocker — No tests**: Added 11 unit tests covering all annotation types, error paths, and edge cases.

**🟡 Should-Fix — expect() panic**: `new()` now accepts `serde_json::Value` directly and returns `Result<Self, SchemaUiError>`. No more panics.

**🟡 Should-Fix — Silent no-op on wrong root name**: `new()` returns `Err(SchemaUiError::RootSchemaNotFound)` upfront. Unknown field names fire `debug_assert!` in dev builds.

**🟡 Should-Fix — Unnecessary roundtrip**: `new()` accepts `Value` directly. All 6 callers updated to use `serde_json::to_value()`.

**🔵 Low — collapsed doc**: Updated doc to clarify group-level semantics and last-write-wins behavior.

**🔵 Low — x-ui:help**: Added `help(text: &str)` method to `FieldUiBuilder`.

**🔵 Nit — PR description**: `.apply()` in the description should be `.annotate()` — will fix in PR description.

**Review by ruokun-niu — APPROVED** [review]

**Review by agentofreality — APPROVED** [review]
